src/stm32/flash.c

```diff
@@ -221,6 +221,10 @@ flash_write_block(uint32_t block_address, uint32_t *data)
     if (need_erase)
         erase_page(page_address);
 
+    // avoid triggering STM32H72xx write security
+    lock_flash();
+    unlock_flash();
+
```

Comment:
Since this is only needed after an erase we should place it within the `need_erase` conditional block.  I also think we should limit to the H7 series to be sure there is no regression on other platforms.
```c
if (need_erase) {
    erase_page(page_address);
    if (CONFIG_MACH_STM32H7) {
        // avoid triggering STM32H72xx write security
        lock_flash();
        unlock_flash();
    }
}
```